File: 2271-rearrange-array-elements-by-sign/2271-rearrange-array-elements-by-sign.java
class Solution {
    public int[] rearrangeArray(int[] nums) {
        if(nums == null || nums.length == 0){
            return nums;
        }
        int[] ans = new int[nums.length];
        int pos = 0;
        int neg = 1;
        for(int i = 0; i<nums.length; i++){
            if(nums[i] < 0){
                ans[neg] = nums[i];
                neg += 2;
            }else {
                ans[pos] = nums[i];
                pos += 2;
            }
        }

        return ans;
    }
}

File: 2271-rearrange-array-elements-by-sign/2271-rearrange-array-elements-by-sign.py
class Solution(object):
    def rearrangeArray(self, nums):
        """
        :type nums: List[int]
        :rtype: List[int]
        """
        result = [0] * len(nums)
        neg = 1
        pos = 0
        
        for i in range(len(nums)):
            if nums[i] < 0:
                result[neg] = nums[i]
                neg = neg+2
            if nums[i] > 0:
                result[pos] = nums[i]
                pos = pos+2
        
        return result